Fronter/Source/UpdateChecker/UpdateChecker.cpp

```diff
@@ -0,0 +1,216 @@
+#include "UpdateChecker.h"
+#include "VersionParser.h"
+#include <set>
+
+
+bool newerVersion(const std::string& currentVersion, const std::set<std::string>& versions)
+{
+	auto newestVersion = currentVersion;
+	auto changed = false;
+	for (const auto& version : versions)
+	{
+		const auto newestDotPos = newestVersion.find('.');
+		const auto versionDotPos = version.find('.');
+		
+		if (std::stoi(version) > std::stoi(newestVersion))
+		{
+			newestVersion = version;
+			changed = true;
+			continue;
+		}
+		if (std::stoi(version) < std::stoi(newestVersion))
+			continue;
+
+		// if parts before dot are equal, compare the rest
+		auto newestRSide = newestDotPos == std::string::npos ? "" : newestVersion.substr(newestDotPos + 1);
+		auto versionRSide = versionDotPos == std::string::npos ? "" : version.substr(versionDotPos + 1);
+		if (version.size() > newestVersion.size())
+		{
+			bool setAfterLoop = true;
+			if (newestDotPos != std::string::npos)
+			{
+				for (unsigned int i = 0; i < newestRSide.size(); ++i)
+				{
+					if (toupper(versionRSide[i]) > toupper(newestRSide[i]))
+					{
+						newestVersion = version;
+						changed = true;
+						setAfterLoop = false;
+						break;
+					}
+					if (toupper(versionRSide[i]) < toupper(newestRSide[i]))
+					{
+						setAfterLoop = false;
+						break;
+					}
+				}
+			}
+			if (setAfterLoop && versionRSide.substr(newestRSide.size(), versionRSide.size() - newestRSide.size()).find_first_not_of('0') != std::string::npos)
+			{
+				newestVersion = version;
+				changed = true;
+			}
+		}
+		else // version shorter or same length as newestVersion
+		{
+			if (versionDotPos != std::string::npos)
+			{
+				for (unsigned int i = 0; i < versionRSide.size(); ++i)
+				{
+					if (toupper(versionRSide[i]) > toupper(newestRSide[i]))
+					{
+						newestVersion = version;
+						changed = true;
+						break;
+					}
+					if (toupper(versionRSide[i]) < toupper(newestRSide[i]))
+						break;
+				}
+			}
+		}
+	}
+
+	if (changed)
+		return true;
+	return false;
+}
+
+
+std::set<std::string> getAllNumberedVersions(std::string& json)
+{
+	std::set<std::string> versions;
+	size_t pos;
+	size_t nameStart;
+	size_t nameEnd = 0;
+	do
+	{
+		pos = json.find("\"name\"", nameEnd);
```

Comment:
I'm torn. On the one hand, that's not how you're supposed to do json, and it's likely really fragile. On the other, it saved us a json library.

Reply:
We could use this: https://github.com/nlohmann/json